### PR TITLE
Make base types appropriate for non-ARM resources

### PR DESCRIPTION
**Pull request by matthchr**
Change type hierarchy so that MetaObject doesn't assume a resource exists in ARM.

In the future, the GenericController must function on types that aren't ARM resources, such as MySQL Users.

### CODE REVIEW

File: v2/internal/reconcilers/arm/client_helpers.go [hunk 1]
```diff
@@ -6,50 +6,10 @@
...
 	"sigs.k8s.io/controller-runtime/pkg/client"
-
-	"github.com/Azure/azure-service-operator/v2/internal/util/kubeclient"
-	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
 )
 
-// CommitObject persists the contents of obj to etcd by using the Kubernetes client.
-// Note that after this method has been called, obj contains the result of the update
-// from APIServer (including an updated resourceVersion).
-func CommitObject(ctx context.Context, kubeClient kubeclient.Client, obj genruntime.MetaObject) error {
```
Comment (theunrepentantgeek) [inline v2/internal/reconcilers/arm/client_helpers.go:22]: Love moving things out of global scope onto a type. :+1:

File: v2/pkg/genruntime/arm_id.go [hunk 2]
```diff
@@ -11,11 +11,25 @@ import (
...
+func GetAndParseResourceID(obj ARMMetaObject) (*arm.ResourceID, error) {
 	resourceID, hasResourceID := GetResourceID(obj)
 	if !hasResourceID {
 		return nil, errors.Errorf("cannot find resource id for obj %s/%s", obj.GetNamespace(), obj.GetName())
 	}
 
 	return arm.ParseResourceID(resourceID)
 }
+
+// TODO: We really want these methods to be on MetaObject itself -- should update code generator to make them at some point
```
Comment (theunrepentantgeek) [inline v2/pkg/genruntime/arm_id.go:23]:
```suggestion
// TODO: We really want these methods to be on ARMMetaObject itself -- should update code generator to make them at some point
```

**Review by theunrepentantgeek — APPROVED** [review]
One minor suggestion to update a comment, otherwise looks good.

File: v2/pkg/genruntime/arm_id.go (hunk 2 above)
Comment (matthchr) [reply v2/pkg/genruntime/arm_id.go:23]: Fixed

**Pull request merged**